### PR TITLE
Support Conditional Clicking in Broker Protection

### PR DESCRIPTION
**Pull request by brianhall**
**Asana Task/Github Issue:** https://app.asana.com/0/72649045549333/1208561106797172/f

## Description
Adds support for conditional clicking in Broker Protection, allowing the click action to evaluate the user's profile in order to determine which element gets clicked.

## Testing Steps

- Run test-suites
- Drag/drop into XCode

## Checklist

*Please tick all that apply:*

- [ ] I have tested this change locally
- [ ] I have tested this change locally in all supported browsers
- [ ] This change will be visible to users
- [ ] I have added automated tests that cover this change
- [ ] I have ensured the change is gated by config
- [ ] This change was covered by a ship review
- [ ] This change was covered by a tech design
- [ ] Any dependent config has been merged

### CODE REVIEW

File: injected/src/features/broker-protection/actions/click.js [hunk 1]
```diff
@@ -63,3 +136,31 @@ function selectRootElement (clickElement, userData, root = document) {
...
+ * 
+ * @param {string} operator 
+ * @returns 
+ */
+export function getComparisonFunction(operator) {
+    switch (operator) {
+      case "==":
+        return (a, b) => a == b;
+      case "===":
+        return (a, b) => a === b;
```
Comment (jonathanKingston) [inline injected/src/features/broker-protection/actions/click.js:150]: I'm guessing you don't need == and === here right?

**Review by jonathanKingston — CHANGES_REQUESTED** [review]
Some eslint warnings here, I assume you're still working. Place it as a draft :).

Is there a task for these changes?

**Comment by brianhall** [comment]
@jonathanKingston Added associated task. Apologies, should have started with draft PR as I was just stashing changes for the weekend.

File: injected/integration-test/test-pages/broker-protection/actions/expectation-actions-conditional-subaction.json [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+          }
+        ],
+        "actions": [
+            {
+                "actionType": "click",
+                "id": "1",
+                "choices": [
+                    { 
+                        "condition": {
+                            "left": "40",
```
Comment (brianhall) [inline injected/integration-test/test-pages/broker-protection/actions/expectation-actions-conditional-subaction.json:20]: This is the test that ties everything together, but using `${age}` here throws an error ("age not found") not sure if that's due to the way the test harness works, or if there's a genuine issue here.

File: injected/src/features/broker-protection/actions/expectation.js [hunk 3]
```diff
@@ -34,7 +34,7 @@ export async function expectation (action, userData, root = document) {
             const result = await execute(subAction, userData, root)
 
             if ('error' in result) {
-                secondaryErrors.push(result.error)
+                secondaryErrors.push(`Sub-action actionID ${subAction.id} error - ${result.error.message}`)
```
Comment (brianhall) [inline injected/src/features/broker-protection/actions/expectation.js:37]: Realized I wasn't reporting the correct error id (or type) in the previous PR.

File: injected/src/features/broker-protection/actions/click.js [hunk 4]
```diff
@@ -63,3 +99,112 @@ function selectRootElement (clickElement, userData, root = document) {
...
 }
+
+/**
+ * Evaluate a comparator and return the appropriate function
+ * @param {string} operator
+ * @returns {(a: any, b: any) => boolean}
+ */
+export function getComparisonFunction (operator) {
+    switch (operator) {
+    case '=':
```
Comment (jonathanKingston) [inline injected/src/features/broker-protection/actions/click.js:110]: I'm still interested why we need all these three? Same for != and !==. '=' is assignment usually too?

Comment (brianhall) [reply injected/src/features/broker-protection/actions/click.js:110]: I don't think it's clear [in the JSON](https://github.com/duckduckgo/content-scope-scripts/pull/1141/files#diff-56d394cac240644f146ac4a55d38e24230eb34131b8aeb6900c3b9b6a8c569efR10) that these operators will behave according to Javascript rules, so I'm mostly just trying to safeguard against someone less familiar with the internals putting `=` when they almost certainly mean `===` (same goes for `!=` and `!==`).

**Review by shakyShane — APPROVED** [review]

**Pull request merged**